File: test-sources/java-deep/src/main/java/player/fr/sii/game/player/Player.java
package main.java.player.fr.sii.game.player;

import main.java.player.fr.sii.game.weapon.Weapon;

public class Player {

    private String name;
    private int power;
    private PlayerType type;
    private Weapon weapon;

    public Player(String pName) {
        this.name = pName;
        this.type = PlayerType.SOLDIER;
        this.weapon = new Weapon("wa");
    }

    public void init(int pPower){
        this.power = pPower;
    }
}

File: demos/typescript/typescript/player/player.ts
import { PLAYER_TYPES } from './player-types';
import { Weapon } from '../weapon/weapon';

export class Player {
    name: string;
    power: number;

    weapon: Weapon;

    type = PLAYER_TYPES.SOLDIER;

    constructor(name: string) {
        this.name = name;
        this.weapon = new Weapon(`weapon-${name}`);
    }

    init(power: number) {
        this.power = power;
    }
}
